File: behavioral_patterns/template_method/java/src/beverageProductionTemplate/Coffee.java
package beverageProductionTemplate;

public class Coffee extends BeverageProductionTemplate {

    @Override
    void mixIngredients() {
        System.out.println("Dripping Coffee through filter");
    }

    @Override
    void addCondiments() {
        System.out.println("Adding Sugar and Milk");
    }
}

File: behavioral_patterns/template_method/typescript/src/Coffee.ts
import {BeverageProductionTemplate} from "./BeverageProductionTemplate";

export class Coffee extends BeverageProductionTemplate {
  protected mixIngredients(): void {
      console.log("Dripping Coffee through filter");
  }

  protected addCondiments(): void {
      console.log("Adding Sugar and Milk");
  }
}